src/main/kotlin/data/persistence/AppPreferences.kt

```diff
@@ -0,0 +1,7 @@
+package data.persistence
+
+// interface to getting user-independent app preferences
+interface AppPreferences {
+    fun getCurrentUserId(): Int // should return an empty string if no current user hash
```

Comment:
you're right that a current id may not exist; but you can't return an empty string if your return type is an Int. This is a scenario where a Int? is valid.

Reply:
Right, I think that comment was left over from when AppPreferences was storing/retrieving the hash string. Is `Int?` better than returning 0, since 0 will never be used as an ID in the database?